Scaffolding with JSON/Jade/Sass

hoichi opened the pull request:
Ok, here's my attempt at a PR. Sorry it took me so long.

A few notes:

* My gulp script builds to `/distr`. I know there's a `/build` directory already but I'm not sure putting .md and html together is a good thing.
* And anyway, the `deploy` gulp task pushes the content of the `/distr` to the root of `gh-pages` branch (so it presumes that branch exist).
* There's a `md-import.js` right in the `/src/data` folder. Messy, and as messy on the inside, but if you switch to maintaining JSON instead of Markdown, you can simply delete everything but the `the-big-data.json`. On the other hand, if you want me to refactor anything, I will.
* The `the-big-data.json` is freshly converted from your latest .md.

Again, if you want me to refactor anything before you accept my PR, I will. I want to do it right ^_^

ericelliott commented:
> My gulp script builds to /distr. I know there's a /build directory already but I'm not sure putting .md and html together is a good thing.

I prefer the more common name, `dist`

> There's a md-import.js right in the /src/data folder. Messy, and as messy on the inside, but if you switch to maintaining JSON instead of Markdown, you can simply delete everything but the the-big-data.json. On the other hand, if you want me to refactor anything, I will.

I like to have a `/scripts` folder for things like this.

hoichi commented:
> I prefer the more common name, dist

It _was_ `dist`. Sorry for the confusion.

> I like to have a /scripts folder for things like this.

Ok, I moved `md-import.js` to `/scripts` and added a task to `package.json`, so you can run `npm run md-import`.

I also renamed the JSON file to `/src/data/essential-javascript-links.json`.

ericelliott commented:
Nice! =)

:+1:
